### ISSUE TITLE
Notification Settings: chosen theme sometime lost

### ISSUE THREAD

**Issue by lukefromdc** [mate-desktop/mate-control-center#222]
Beginning with the Feb 4 changes to mate-control-center, I noticed an issue where sometimes the notification theme would return to "standard" and if I opened the control center "Notification Settings" window, the label in the "Theme" button would be blank. Setting it back to "Coco" would bring up that theme, but sometimes it would be "forgotten" and blanked again later.

My theme does not explicitly set a notification theme, I have always done this from the control center and not had this problem since my days of running gtk2 builds of mate 1.8. Changing to one of the MATE themes, then back, or rebuilding and reinstalling mate-control-center has occasionally done this in the past, but today I had a string of notifications form Audacious (one for each new song) suddenly change back to the "standard" theme after coming up fine in Coco for several songs, and no changes to themes or the control center.

**Comment by monsta** [comment]
>  if I opened the control center "Notification Settings" window, the label in the "Theme" button would be blank

I saw that a few months ago (with GTK+ 3.18). The settings tool is a part of mate-notification-daemon package BTW :smile:

**Comment by lukefromdc** [comment]
Is this caused by themes not explicitly setting a notification theme, or 
do you get that even in themes that do so?

**Comment by monsta** [comment]
I can always reproduce that blank label by first setting theme to BlackMATE in mate-appearance-properties, then launching mate-notification-properties. Actually, I can see it in a GTK+2 build of MATE with mate-notification-daemon built with GTK+3 (version 3.14.5 here).

**Comment by raveit65** [comment]
The issue with BlackMate theme is different.
I confirm that sometimes the notification theme reset to standard theme, but only with gtk+-3.19.x.
I didn't noticed that with 3.16 here.

**Comment by raveit65** [comment]
Btw. we have to check if the dconf key is changed too.

**Comment by lukefromdc** [comment]
This is not a new bug, as I said I used to see this in mate 1.8 with GTK 2

**Comment by monsta** [comment]
When setting theme to BlackMATE, notification theme gets reset to "Coco" in ```org.mate.NotificationDaemon``` - but the correct key value for that theme is "coco".

**Comment by monsta** [comment]
Fix for that issue: https://github.com/mate-desktop/mate-themes/pull/103

**Comment by monsta** [comment]
> I noticed an issue where sometimes the notification theme would return to "standard" and if I opened the control center "Notification Settings" window, the label in the "Theme" button would be blank.

BTW, @raveit65 is right - check "theme" key in org.mate.NotificationDaemon schema when it happens...

**Comment by lukefromdc** [comment]
I just found that the index.theme file in my own theme was corrupted and ignored. Never did set a notification theme there as the theme originally predated MATE. Setting it there seems to keep it from disappearing, along with fixing the same file in BlueMenta, from which I got the idea to check my own.
Will check this now in Audacious with notifications enabled to see if they are now consistant.

**Comment by raveit65** [comment]
Yeah, but setting a notification theme in index.theme is not really a solution.
Which theme should displayed is set in a gsettings key and nothing should override this setting.

**Comment by monsta** [comment]
Well... BlackMATE seems to override that setting with "coco" :smile:

**Comment by raveit65** [comment]
> Well... BlackMATE seems to override that setting with "coco" :smile:

You're kidding :palm_tree: 
I meant that nothing else should override this settings.
Btw, i don't see that BlackMate with latest commit change the notification theme to coco in gtk3 environment (3.16)
